Add more custom parameters for launching RHEL machines on AWS

gpei opened the pull request.

jianlinliu commented on ci-operator/step-registry/workers-rhel/aws-provision/workers-rhel-aws-provision-commands.sh:
```diff
@@ -128,11 +128,16 @@ cat > create_machineset.yaml <<-'EOF'
     machineset: "{{ machineset_obj | combine(dict_edit, recursive=True) }}"
   vars:
     ssh_key_name: "{{ lookup('env', 'SSH_KEY_NAME') }}"
+    machineset_replicas: "{{ lookup('env', 'RHEL_WORKER_COUNT_PER_ZONE') }}"
```
I did not find out the ZONE setting in machineset yaml, you can help point me there.

gpei replied:
@jianlinliu In this AWS RHEL machine provisioning process, it's updating the existing RHCOS worker machineset file one by one and using them to create RHEL machines, because the each RHCOS worker `machineset` itself was created per Zones, so I choose this name. 

For example, if there're three RHCOS machinesets in the beginning, 
```
[root@preserve-gpei-worker k_files]#   oc get machinesets -n openshift-machine-api
NAME                               DESIRED   CURRENT   READY   AVAILABLE   AGE
gpei-415-wc42b-worker-us-east-2a   1         1         1       1           123m
gpei-415-wc42b-worker-us-east-2b   1         1         1       1           123m
gpei-415-wc42b-worker-us-east-2c   1         1         1       1           123m
```

In the previous code, the RHEL worker number would be the same "3" as RHCOS machines, we have no way to customize the RHEL worker number but some team member is requesting this. So if we use this `RHEL_WORKER_COUNT_PER_ZONE` parameter to specify the `machineset_replicas`, such as `RHEL_WORKER_COUNT_PER_ZONE=3`, we'll get 3*3=9 RHEL workers.

jianlinliu replied:
Got it, thanks for the detailed explanation.

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

jianlinliu commented:
/lgtm

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

gpei commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-disconnected-workers-rhcos-rhel8-f14

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

gpei commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-disconnected-workers-rhcos-rhel8-f14

gpei commented:
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws

gpei commented:
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-disconnected-workers-rhcos-rhel8-f14

gpei commented:
@jianlinliu please help to check again, in the previous code, I was using 
`replicas: "{{ machineset_replicas }}"` to set the replicas of each machineset, but it turns out there's a limitation from ansible side is the output of Jinja template is always _string_, no matter we force convert it to _int_ or not(https://www.freekb.net/Article?id=2570), and the machineapi is expecting the `replicas` field to be an _int_.

```
        "Error from server: error when creating \"STDIN\": admission webhook \"default.machineset.machine.openshift.io\" denied the request: json: cannot unmarshal string into Go struct field MachineSetSpec.spec.replicas of type int32"
```

A test script on this
```
[root@preserve-gpei-worker ansible_test]# cat test.yaml 
---
- hosts: localhost
  vars:
    machineset_replicas: "{{ lookup('env', 'RHEL_WORKER_COUNT_PER_ZONE') }}"
  tasks:
  - set_fact:
      replicas: "{{ machineset_replicas | int }}"

  - debug: var=replicas

  - debug:
      msg: "The var type of 'machineset_replicas' is {{ machineset_replicas | type_debug }}"

  - debug:
      msg: "The var type of 'replicas' is {{ replicas | type_debug }}"

  - debug:
      msg: "The var type of 'replicas | int' is {{ replicas | int | type_debug }}"
```

The output:

```
[root@preserve-gpei-worker ansible_test]# export RHEL_WORKER_COUNT_PER_ZONE=10
[root@preserve-gpei-worker ansible_test]# ansible-playbook test.yaml 
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ***********************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************
ok: [localhost]

TASK [set_fact] ************************************************************************************************************
ok: [localhost]

TASK [debug] ***************************************************************************************************************
ok: [localhost] => {
    "replicas": "10"
}

TASK [debug] ***************************************************************************************************************
ok: [localhost] => {
    "msg": "The var type of 'machineset_replicas' is AnsibleUnsafeText"
}

TASK [debug] ***************************************************************************************************************
ok: [localhost] => {
    "msg": "The var type of 'replicas' is AnsibleUnsafeText"
}

TASK [debug] ***************************************************************************************************************
ok: [localhost] => {
    "msg": "The var type of 'replicas | int' is int"
}

PLAY RECAP *****************************************************************************************************************
localhost                  : ok=6    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

So I changed to use `oc scale --replicas={{ machineset_replicas }} machineset ` command to set the RHEL worker number of each machineset, seen from the passed rehearse job https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/47579/rehearse-47579-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-aws/1746807314981064704 , it works as expected. 

```
TASK [Get the RHEL machineset replicas setting] ********************************
task path: /usr/share/ansible/openshift-ansible/create_machineset.yaml:44
ok: [localhost] => {
    "ansible_facts": {
        "machineset_replicas": "3"
    },
    "changed": false
}

TASK [Scale the RHEL machineset replicas as needed] ****************************
task path: /usr/share/ansible/openshift-ansible/create_machineset.yaml:48
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: default
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/.ansible-tmp `"&& mkdir "` echo /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002 `" && echo ansible-tmp-1705310221.373795-220-185404575698002="` echo /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002 `" ) && sleep 0'
Using module file /usr/lib/python3.9/site-packages/ansible/modules/command.py
<127.0.0.1> PUT /opt/app-root/src/.ansible/tmp/ansible-local-652chyx7nh/tmp31gxpet3 TO /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002/AnsiballZ_command.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002/ /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002/AnsiballZ_command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3.9 /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002/AnsiballZ_command.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /tmp/.ansible-tmp/ansible-tmp-1705310221.373795-220-185404575698002/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "attempts": 1,
    "changed": true,
    "cmd": [
        "oc",
        "scale",
        "--replicas=3",
        "machineset",
        "ci-op-21gmhbk2-066bf-q8np2-worker-us-east-1a-rhel",
        "--kubeconfig=/tmp/secret/kubeconfig",
        "--namespace=openshift-machine-api",
        "--output=json"
    ],
```

Later I found there's an environment var named `ANSIBLE_JINJA2_NATIVE` may help solving such issue,
https://docs.ansible.com/ansible/latest/reference_appendices/config.html#envvar-ANSIBLE_JINJA2_NATIVE, 
it preserves variable types during template, but it's requiring `Jinja 2.10 and above` version to use it, 
```
[root@preserve-gpei-worker ansible_test]# export ANSIBLE_JINJA2_NATIVE=True
[root@preserve-gpei-worker ansible_test]# ansible-playbook create-machines.yaml -v
[WARNING]: jinja2_native requires Jinja 2.10 and above. Version detected: 2.7.2. Falling back to default.
```
Since this is not something critical function to be updated, so I don't plan to update the Jinja2 version of the images anymore, just keep using the current logic.

jianlinliu commented:
/lgtm

gpei commented:
/pj-rehearse ack
